File: 8KYU/Count of Positive Sum of Negatives/solution.rb
def count_positives_sum_negatives(lst)
    if lst.length() == 0
      return []
    else
      pos = (lst.select do |n| n > 0 end).length()
      neg = (lst.select do |n| n < 0 end).sum()
      
      return [pos, neg]
    end
end
  

File: 8KYU/Count of Positive Sum of Negatives/solution.c
#include <stddef.h>

void count_positives_sum_negatives(
  int *values, size_t count, int *positivesCount, int *negativesSum) 
{
  int cnt = 0;
  int sum = 0;
  
  for(size_t i = 0; i < count; i++) {
    if(values[i] <= 0)
      sum += values[i];
    else
      cnt++;
  }
  
  *positivesCount = cnt;
  *negativesSum = sum;
}  